### PR TITLE
Explore page

### PR DESCRIPTION
**Pull request by harshitpaliwal95**
## feat
Explore page
- new posts filter
- older posts filter
- new connection (working)
- upload avatar

### CODE REVIEW

File: src/components/modal/uploadButton.js [hunk 1]
```diff
@@ -18,20 +18,23 @@ export const UploadButtons = ({ setActarUploading }) => {
...
+      if (data) {
+        const { publicURL, error } = supabase.storage
+          .from("avatars")
+          .getPublicUrl(fileName);
+
+        setAvatar(publicURL);
+        if (error) {
+          throw error;
+        }
+        console.log(data);
```
Comment (xii-vi) [inline src/components/modal/uploadButton.js]: remove console log

File: src/components/modal/userEditModal.js [hunk 2]
```diff
@@ -25,26 +26,30 @@ const style = {
...
-          username: userData.userName,
-          user_bio: userData.userBio,
-          website: userData.userWebsite,
+          username: userData.userName ?? profile.userName,
+          user_bio: userData.userBio ?? profile.userBio,
+          website: userData.userWebsite ?? profile.userPortfolio,
+          avatar_url: avatar ?? profile.userAvtar,
         })
         .eq("id", auth.userID);
       console.log(data);
```
Comment (xii-vi) [inline src/components/modal/userEditModal.js]: here too

**Review by xii-vi — APPROVED** [review]
The code Structure is clean.
consistent design and great UI.
left a few comments check them out once before merging.

**Review by Sharmaryan — APPROVED** [review]
- I am unable to login, look into that part
- Code is clean

**Pull request merged**